routemaster/state_machine/api.py

```diff
@@ -101,6 +101,44 @@ def create_label(app: App, label: LabelRef, metadata: Metadata) -> Metadata:
     return metadata
 
 
+def restore_label_and_restart(
+    app: App,
+    label: LabelRef,
+    replacement_metadata: Metadata,
+) -> Metadata:
+    """
+    Restores a label that was previously deleted.
+
+    We replace the metadata entirely rather than merging it, so as to prevent
+    old and stale data remaining in the system longer than it needs to. This
+    function restarting the label into a completely fresh state cuts down on
+    the number of possible states a label can exist in.
+    """
+    state_machine = get_state_machine(app, label)
+
+    row = lock_label(app, label)
+
+    # Record the label as having been restored and add its initial metadata
+    if not row.deleted:
+        raise LabelAlreadyExists(f"Label {label} is not marked as deleted!")
+
+    row.metadata = replacement_metadata
+    row.deleted = False
+
+    # Add a history entry for the deletion.
+    current_state = get_current_state(app, label, state_machine)
+
+    if current_state is not None:
+        raise LabelAlreadyExists(f"Deleted label {label} has current state!")
+
+    row.history.append(History(
+        old_state=None,
+        new_state=state_machine.states[0].name,
+    ))
+
```

Comment:
In order to restart the label within the state machine, I think we also need to `process_transitions` here, just like we do in `create_label`. Is that not the case?